subsys/bluetooth/mesh/provisioner.c

```diff
@@ -411,6 +411,13 @@ static void prov_dh_key_gen(void)
 	sys_memcpy_swap(remote_pk_le, remote_pk, 32);
 	sys_memcpy_swap(&remote_pk_le[32], &remote_pk[32], 32);
 
+	if (!memcmp(&bt_mesh_prov_link.conf_inputs[17],
+		    &bt_mesh_prov_link.conf_inputs[81], 64)) {
```

Comment:
It might make the code slightly more readable if you used friendly names for these two pointers rather than "magic" indices to the `conf_inputs` array. You already have `remote_pk` that you can use as the second parameter, so I'd consider adding a second `local_pk` pointer to `&conf_inputs[17]` and pass these to `memcmp`

Reply:
I was just putting on my list to make this a union of multiple arrays and conf_inputs :smile:

Reply:
Done